bluebutton-data-pipeline-rif-extract/src/main/java/gov/hhs/cms/bluebutton/datapipeline/rif/extract/RifFilesProcessor.java

```diff
@@ -313,13 +315,15 @@ private static PartDEventRow buildPartDEvent(CSVRecord csvRecord) {
 		pdeRow.nationalDrugCode = csvRecord.get(PartDEventRow.Column.PROD_SRVC_ID.ordinal());
 		pdeRow.planContractId = csvRecord.get(PartDEventRow.Column.PLAN_CNTRCT_REC_ID.ordinal());
 		pdeRow.planBenefitPackageId = csvRecord.get(PartDEventRow.Column.PLAN_PBP_REC_NUM.ordinal());
-		pdeRow.compoundCode = Integer.parseInt(csvRecord.get(PartDEventRow.Column.CMPND_CD.ordinal()));
+		pdeRow.compoundCode = CompoundCode
+				.parseRifValue(Integer.parseInt(csvRecord.get(PartDEventRow.Column.CMPND_CD.ordinal())));
 		pdeRow.dispenseAsWrittenProductSelectionCode = csvRecord.get(PartDEventRow.Column.DAW_PROD_SLCTN_CD.ordinal());
 		pdeRow.quantityDispensed = new BigDecimal(csvRecord.get(PartDEventRow.Column.QTY_DPSNSD_NUM.ordinal()));
 		pdeRow.daysSupply = Integer.parseInt(csvRecord.get(PartDEventRow.Column.DAYS_SUPLY_NUM.ordinal()));
 		pdeRow.fillNumber = Integer.parseInt(csvRecord.get(PartDEventRow.Column.FILL_NUM.ordinal()));
 		pdeRow.dispensingStatuscode = parseOptCharacter(csvRecord.get(PartDEventRow.Column.DSPNSNG_STUS_CD.ordinal()));
-		pdeRow.drugCoverageStatusCode = csvRecord.get(PartDEventRow.Column.DRUG_CVRG_STUS_CD.ordinal()).charAt(0);
+		pdeRow.drugCoverageStatusCode = DrugCoverageStatus
+				.parseRifValue(csvRecord.get(PartDEventRow.Column.DRUG_CVRG_STUS_CD.ordinal()).charAt(0));
```

Comment:
I'd remove the `.charAt(...)` call here, and have `DrugCoverageStatus.parseRifValue(...)` go boom if there's more than one character.